### PR TITLE
NOJIRA: Update opensaml dependency w/ minor clean up

### PR DESCRIPTION
**Pull request by mmoayyed**
- Update OpenSAML dependency to the latest available version, 3.4.4
- Add relevant repositories to locate and resolve artifacts
- Remove unused class (which can in fact be turned into a default interface method if necessary)
- Scanned for usages of deprecated content in the new OpenSAML version; none were found.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1800,6 +1800,16 @@ under the License.
       <id>jitpack.io</id>
       <url>https://jitpack.io</url>
     </repository>
+    <repository>
```
[the remark this answers is not in the document]
Comment (mmoayyed) [reply pom.xml]: The canonical repository for resolving opensaml dependencies is the shibboleth repository. The development team will never publish to maven central, and are quite against doing so for security purposes. Without this, I wasn't able to resolve and build with the new version.

[the remark this answers is not in the document]
Comment (ilgrosso) [reply pom.xml]: Hmm, I suspect that the people at ASF Maven would seriously disagree on the concept of "security" here... anyway, was this a recent decision? I can see up to 3.4.3 on Maven Central: https://search.maven.org/artifact/org.opensaml/opensaml-parent/3.4.3/pom

[the remark this answers is not in the document]
Comment (mmoayyed) [reply pom.xml]: Not a recent decision, no. The decision I think started around the invention of OpenSAML v3. The development team, IIRC, is quite explicit about not publishing to maven central, and do say that "if you do find stuff on central, it's not by us and we do not endorse it at all".

Reference: https://wiki.shibboleth.net/confluence/display/DEV/Use+of+Maven+Central

So, maybe artifacts are found on Central and maybe they are published by some entity perfectly good enough by GA; but either way, I am not quite sure we should be using them here, given the above link.

[the remark this answers is not in the document]
Comment (ilgrosso) [reply pom.xml]: Thanks for bringing this up: given this decision, then let's add the repository.

**Comment by mmoayyed** [comment]
Would you prefer if we changed `RequestedAuthnContextProvider` to:

```java
public interface RequestedAuthnContextProvider {

    default RequestedAuthnContext provide() { return null; }
}
```

**Comment by ilgrosso** [comment]
I am afraid it's not that simple; as many other interfaces throughout the whole codebase, this one is meant to be managed via an [implementation](https://ci.apache.org/projects/syncope/master/reference-guide.html#implementations).

FYI the application logic for this implementation is located at https://github.com/apache/syncope/blob/master/ext/saml2sp/logic/src/main/java/org/apache/syncope/core/logic/SAML2SPLogic.java#L298-L307

**Comment by mmoayyed** [comment]
Makes sense, sure. I'll revert.

**Pull request merged**